components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt

```diff
@@ -4,28 +4,45 @@
 
 package mozilla.components.service.fxa
 
+import androidx.lifecycle.LifecycleOwner
 import kotlinx.coroutines.async
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Deferred
 import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.plus
 import mozilla.appservices.fxaclient.FirefoxAccount as InternalFxAcct
 import mozilla.appservices.fxaclient.FxaException.Unauthorized as Unauthorized
 
 import mozilla.components.concept.sync.AccessTokenInfo
-import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceConstellation
+import mozilla.components.concept.sync.DeviceConstellationObserver
+import mozilla.components.concept.sync.DeviceEvent
+import mozilla.components.concept.sync.DeviceEventOutgoing
+import mozilla.components.concept.sync.DevicePushSubscription
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.concept.sync.DeviceEventsObserver
 import mozilla.components.concept.sync.Profile
 import mozilla.components.concept.sync.StatePersistenceCallback
+import mozilla.components.concept.sync.TabData
+import mozilla.components.service.fxa.manager.DeviceManagerProvider
+import mozilla.components.service.fxa.manager.PollingDeviceManager
 import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.base.observer.Observable
+import mozilla.components.support.base.observer.ObserverRegistry
 
 typealias PersistCallback = mozilla.appservices.fxaclient.FirefoxAccount.PersistCallback
 
 /**
  * FirefoxAccount represents the authentication state of a client.
  */
 @Suppress("TooManyFunctions")
-class FirefoxAccount internal constructor(private val inner: InternalFxAcct) : OAuthAccount {
+class FirefoxAccount internal constructor(
+    private val inner: InternalFxAcct
+) : DeviceConstellation, Observable<DeviceEventsObserver> by ObserverRegistry() {
```

Comment:
I think here it's even clearer that there's something wrong with the `is a` relationship. Why is a `FirefoxAccount` a `DeviceConstellation`? I would definitely switch that to `has a` and use composition.